src/ucp/core/ucp_worker.c

```diff
@@ -3120,8 +3149,13 @@ void ucp_worker_keepalive_add_ep(ucp_ep_h ep)
     ucs_trace("ep %p flags 0x%x: adding to keepalive lane_map 0x%x", ep,
               ep->flags, ucp_ep_config(ep)->key.ep_check_map);
     uct_worker_progress_register_safe(worker->uct,
-                                      ucp_worker_keepalive_progress, worker,
-                                      UCS_CALLBACKQ_FLAG_FAST,
+                                      !(worker->context->config.features &
+                                        UCP_FEATURE_WAKEUP) ?
+                                      ucp_worker_keepalive_progress :
+                                      /* Do keepalive directly, no need to skip
+                                       * iterations */
+                                      ucp_worker_do_keepalive_progress,
```

Comment:
maybe a call to ucp_worker_arm will make sure the next keepalive progress will not skip iterations? assuming worker_arm is done every time before a potentially long sleep.
and this way, we "don't skip iterations" only when wakeup feature is really used, and not just requested

Reply:
also need to add comment that the reason we do it is to not miss keepalive rounds after a long time without calling progress

Reply:
good idea, done